andela/ah-backend-spaces
# #159082382 Add users comment on Articles.

### What does this PR do?
- Adds commenting functionality on Authors Haven to a particular article. 

### Description of Task to be completed?
- A user is now able to comment on a particular article or start a thread on a comment to an article.

### How should this be manually tested?
- In the case of creating a parent comment, create a request in the format below.
```
{
  "comment": {
    "body": "His name was my name too.",
  }
}
```
- As for a child comment i.e. a comment that is a thread of another. Create a request in the form
```
{
  "comment": {
    "body": "His name was my name too.",
    "parent_id": 1
  }
}
```

- The result should be shown as below.
```
{
    "comment": {
        "body": "His name was my name too.",
        "article_id": 1,
        "author": 1,
        "created_at": "2018-08-09T16:53:53.621732Z",
        "message": "Comment created successfully."
    }
}
```
- Send the request to the API route `/api/articles/<int: article_id>/comment/`
- Note: This route is only accessible by autheticated users .i.e. users with a valid auth token.

### Any background context you want to provide?
- I created a database table `Comments` to hold parent comments.
- I created another database `ChildComment` table that inherits a foreign key from the parent `Comments` table. Such comments are thread to another.

### What are the relevant pivotal tracker stories?
- [#159082382](https://www.pivotaltracker.com/story/show/159082382)
